Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2008-01-18
Initial Package Version: 1.5
Upstream Status: Not Submitted
Origin: Fedora's sysklogd-1.4.1-disable__syslog_chk.patch
Description: When compiling Sysklogd with -D_FORTIFY_SOURCE=2 we will
get errors like "conflicting types for '__syslog_chk'". This patch
works around that problem.

--- sysklogd-1.4.1/syslog.c.orig	2001-07-09 00:17:22 -0400
+++ sysklogd-1.4.1/syslog.c	2005-08-24 16:47:46 -0400
@@ -54,11 +54,8 @@
 #include <sys/file.h>
 #include <sys/signal.h>
 #include <sys/syslog.h>
-#if 0
-#include "syslog.h"
-#include "pathnames.h"
-#endif
-
+#undef   syslog
+#undef   vsyslog
 #include <sys/uio.h>
 #include <sys/wait.h>
 #include <netdb.h>
--- sysklogd-1.4.1/klogd.h.orig	2000-12-11 14:46:07 -0500
+++ sysklogd-1.4.1/klogd.h	2005-08-24 17:08:14 -0400
@@ -30,7 +30,9 @@
 #include <stdio.h>
 #include <syslog.h>
 #include <string.h>
 
+#undef syslog
+#undef vsyslog
 
 /* Function prototypes. */
 extern int InitKsyms(char *);
